api_spec/src/http_api_spec_parser_impl.cc

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "http_api_spec_parser_impl.h"
+
+using ::istio::mixer::v1::Attributes;
+using ::istio::mixer::v1::config::client::HTTPAPISpec;
+using ::istio::mixer::v1::config::client::HTTPAPISpecPattern;
+
+namespace istio {
+namespace api_spec {
+
+HttpApiSpecParserImpl::HttpApiSpecParserImpl(const HTTPAPISpec& api_spec)
+    : api_spec_(api_spec) {
+  PathMatcherBuilder<const Attributes*> pmb;
+  for (const auto& pattern : api_spec_.patterns()) {
+    if (pattern.pattern_case() == HTTPAPISpecPattern::kUriTemplate) {
+      if (!pmb.Register(pattern.http_method(), pattern.uri_template(),
+                        std::string(), &pattern.attributes())) {
+        // TODO: log error message.
```

Comment:
Why would this fail? Any reason we can't log the message now instead of waiting?

Reply:
Currently, mixer client library doesn't have log utility.   it only has std::cerr.   In the future, I plan to provide a hook to log to platform logging, such as Envoy logging.

Reply:
Can we add `std:err` for now?

Reply:
Done.   use  GOOGLE_LOG from protobuf library.